src/nncf/experimental/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Iterable, Optional
+
+import torch
+
+from nncf import AdvancedCompressionParameters
+from nncf import BackupMode
+from nncf import CompressionFormat
+from nncf import CompressWeightsMode
+from nncf import Dataset
+from nncf import IgnoredScope
+from nncf import SensitivityMetric
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.logging import nncf_logger
+from nncf.common.tensor_statistics.statistic_point import StatisticPointsContainer
+from nncf.common.utils.backend import BackendType
+from nncf.experimental.quantization.quantizer import Quantizer
+from nncf.quantization.algorithms.algorithm import Algorithm
+from nncf.quantization.algorithms.weight_compression.algorithm import WeightCompression as OriginalWeightCompression
+
+
+class WeightsCompression(Algorithm):
+    """
+    Post-training Weight Compression algorithm implementation.
+
+    Compresses weights of Linear and Embedding layers to 8-bit integer or
+    to 4-bit integer/float depending on mode, ratio and group size.
+    """
+
+    def __init__(
+        self,
+        mode: CompressWeightsMode,
+        quantizer: Quantizer,
+        ratio: float,
+        group_size: int,
+        ignored_scope: IgnoredScope,
+        all_layers: bool,
```

Comment:
Please remove `all_layers` parameter from this algorithm as it is defined solely by the quantizer. Use `all_layers=True` value and add a comment to this line like `# This param is specified by the quantizer and does not affect weight compression`

OR

If it possible (and easy), retrieve this param from the quantizer and pass to the algorithm